lib/Doctrine/DBAL/Schema/Index.php

```diff
@@ -336,4 +341,17 @@ private function samePartialIndex(Index $other)
 
         return ! $this->hasOption('where') && ! $other->hasOption('where');
     }
+
+    /**
+     * Returns whether the index has the same column lengths as the other
+     */
+    private function hasSameColumnLengths(self $other) : bool
+    {
+        $filter = static function (?int $length) : bool {
+            return $length !== null;
+        };
+
+        return array_filter($this->options['lengths'] ?? [], $filter)
```

Comment:
Does the order inside `lengths` affect the results? Also, is it always guaranteed to be an `array`?

Reply:
The order of lengths is expected to be the same as the order of columns, lengths should be an array:

https://github.com/doctrine/dbal/blob/e37fc54d91a9bad1a7579520f2ee63c90e9099e3/lib/Doctrine/DBAL/Schema/Index.php#L101-L107

Reply:
Aha, that clears it up :+1: